### ISSUE TITLE
Add doc for sjs-bundler/module users

### ISSUE THREAD

**Issue by japgolly** [japgolly/scalajs-react#343]
Due to https://github.com/scala-js/scala-js/issues/2797 I cannot support both script-style (`@JSName`/`@JSGlobal`) and module-style (`@JSImport`) in this library.

I'll keep it as script style for now, probably until SJS 1.0 is released.

In order to use scalajs-react with scalajs-bundler / module-style, the following webpack config is required:
```js
module: {
  rules: [{
      test: require.resolve('react'),
      use: [{loader: 'expose-loader', options: 'React'}]
    }, {
      test: require.resolve('react-dom'),
      use: [{loader: 'expose-loader', options: 'ReactDOM'}]
    }],
  },
  entry: ['react', 'react-dom'],
}
```

There's also a demo repo where I have this working with some other features as I experiment with webpack + Scala.JS: https://github.com/japgolly/misc/tree/webpack

Put all this ↑ info in the docs for scalajs-react 1.0.

**Comment by japgolly** [comment]
Oh and btw, there's a branch `topic/scalajs-bundler` which hasn't been merged. It all looks fine but there's some weird timeout in the tests.

**Comment by dispalt** [comment]
You can also do something like...  I haven't done it in a while, so it's probably close to correct.  If you don't want to mess with webpack configs (which breaks the best feature, `enableReloadWorkflow`, imo).

```scala
@JSImport("React", JSImport.Default)
@js.native
object RStub extends js.Object

def main(): Unit = {
  global.React = RStub
}
```

**Comment by japgolly** [comment]
Oh that's cool, no need to manually edit webpack config, thanks @dispalt. I'll add that to the doc.

**Comment by dispalt** [comment]
I should make sure it works as expected.  For production, I actually use it for css styles too (sorry I am trying to convince you it's awesome).

I probably don't need the double elidable + tree shaking, but this is what I use.

```scala

@JSImport("!style-loader!css-loader!less-loader!./css/main/index.less", JSImport.Default)
@js.native
object CssLoader extends js.Object

@elidable(elidable.INFO)
private def loadCss(): Unit = { CssLoader; () }

def main() = {

   if (scala.scalajs.LinkingInfo.developmentMode) {
      loadCss
    }
}
```

**Comment by japgolly** [comment]
Hehe don't worry, I'm already convinced that it's awesome! I'll definitely be using modules & webpack for my main work project! I've decided that scalajs-bundler isn't right now my project just now but I'll still make sure there are instructions for that too. @dispalt if you use scalajs-bundler, if you could review the instructions when I write for it, it would be appreciated :)

**Comment by japgolly** [comment]
Btw, I got an even shorter version, no webpack config required 😆 

```scala
@JSImport("expose-loader?React!react", JSImport.Namespace)
@js.native
object React extends js.Any
React

@JSImport("expose-loader?ReactDOM!react-dom", JSImport.Namespace)
@js.native
object ReactDOM extends js.Any
ReactDOM
```

**Comment by dispalt** [comment]
Oh that's even better.

**Comment by japgolly** [comment]
Closed via 5169e79a83f7c959f29194f2071f2063ba7846c1
Not huge but it'll have to do
